processing/src/main/java/org/apache/druid/query/topn/TopNQuery.java

```diff
@@ -97,9 +99,18 @@ public TopNQuery(
             : postAggregatorSpecs
     );
 
+    final long expectedAllocBytes = aggregatorSpecs.stream().mapToLong(AggregatorFactory::getMaxIntermediateSizeWithNulls).sum();
+    final long maxAggregatorHeapSizeBytes = this.context().getLong(QueryContexts.MAX_TOP_N_AGGREGATOR_HEAP_SIZE_BYTES, TopNQueryConfig.DEFAULT_MAX_AGGREGATOR_HEAP_SIZE_BYTES);
+    this.aggregatorHelper = new TopNAggregatorResourceHelper(expectedAllocBytes, new TopNAggregatorResourceHelper.Config(maxAggregatorHeapSizeBytes));
+
     topNMetricSpec.verifyPreconditions(this.aggregatorSpecs, this.postAggregatorSpecs);
   }
 
+  public TopNAggregatorResourceHelper getAggregatorHelper()
```

Comment:
If the code is changed like I suggested earlier, then this method can be removed.